tsk/fs/ntfs.c

```diff
@@ -1360,18 +1367,29 @@ ntfs_attr_walk_special(const TSK_FS_ATTR * fs_attr,
                     size_t i;
 
                     // decompress the unit
-                    if (ntfs_proc_compunit(ntfs, &comp, comp_unit,
+                    uint8_t reset_uncompress_for_init_size = 0;
+                    if (!init_size_reached) {
+                        if (ntfs_proc_compunit(ntfs, &comp, comp_unit,
                             comp_unit_idx)) {
-                        tsk_error_set_errstr2("%" PRIuINUM " - type: %"
-                            PRIu32 "  id: %d Status: %s",
-                            fs_attr->fs_file->meta->addr, fs_attr->type,
-                            fs_attr->id,
-                            (fs_attr->fs_file->meta->
-                                flags & TSK_FS_META_FLAG_ALLOC) ?
-                            "Allocated" : "Deleted");
-                        free(comp_unit);
-                        ntfs_uncompress_done(&comp);
-                        return 1;
+                            tsk_error_set_errstr2("%" PRIuINUM " - type: %"
+                                PRIu32 "  id: %d Status: %s",
+                                fs_attr->fs_file->meta->addr, fs_attr->type,
+                                fs_attr->id,
+                                (fs_attr->fs_file->meta->
+                                    flags & TSK_FS_META_FLAG_ALLOC) ?
+                                "Allocated" : "Deleted");
+                            free(comp_unit);
+                            ntfs_uncompress_done(&comp);
+                            return 1;
+                        }
+
+                        // handle the initialized size
+                        int64_t remanining_init_size = fs_attr->nrd.initsize - off;
+                        if (remanining_init_size < comp.buf_size_b) {
```

Comment:
I don't understand this check.  Is your goal to provide zeros for everything past the initialized area?

Reply:
Yes - the intention is to zero-fill after the init size is reached, to match Windows' behavior for the file.